### PR TITLE
Added type definitions for react-collapse

### PR DESCRIPTION
**Pull request by Kimahriman**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/react-collapse/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// Definitions by: Adam Binford <https://github.com/Kimahriman>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+import * as React from 'react';
+
+export interface CollapseProps extends React.HTMLProps<Collapse> {
+    isOpened: boolean;
+    children: React.ReactNode;
+
+    springConfig?: { [key: string]: number }; // react-motion doesn't export the config interface
```
Comment (bodhi) [inline types/react-collapse/index.d.ts]: > react-motion doesn't export the config interface

If I'm not mistaken, it exports it here:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6e3c284a47b9e9aeb925e6488e8825bc13808057/types/react-motion/index.d.ts#L11

Comment (bodhi) [reply types/react-collapse/index.d.ts]: Ah, I read that type wrong 😬 

What works for me is:

```
import { OpaqueConfig } from "react-motion"
// ...
springConfig?: OpaqueConfig
```

but I'm only passing react-motion presets to react-collapse.

Comment (Kimahriman) [reply types/react-collapse/index.d.ts]: Yeah neither are quite exactly right. I wasn't sure whether to redefine `SpringHelperConfig` completely or not, and just decided not to.

Comment (bodhi) [reply types/react-collapse/index.d.ts]: Aha. The problem I ran into is that passing react-motion presets to react-collapse gave me a type error...

(Thanks for making the type definition though, by the way!)

**Pull request merged**